File: GGJ2015/Assets/Scripts/Texto.js
#pragma strict
//script inserido nos textos do jogo
//simula alguem digitando
//peguei da internet

var currentPosition : int = 0;
var Delay : float = 0.1;  // 10 characters per sec.
var Text : String = "";
var additionalLines : String[];
 
function WriteText(aText : String) {
    guiText.text = "";
    currentPosition = 0;
    Text = aText;
}
 
function Start(){
	
    	for ( var S : String in additionalLines )
        	Text += "\n" + S;
    	while (true){
        	if (currentPosition < Text.Length)
            	guiText.text += Text[currentPosition++];
        	yield WaitForSeconds (Delay);
    				}

}

@script RequireComponent(GUIText)

File: GGJ2015/Assets/Scripts/Texto.cs
using UnityEngine;
using System.Collections;

public class Texto : MonoBehaviour {

    int currentPosition = 0;
    float Delay = 0.1f;
    string Text = "";
    string[] additionalLines;

	// Use this for initialization
	IEnumerator Start () {
	    foreach(string s in additionalLines){
            Text += "\n" + s;
        }
        while (true)
        {
            if (currentPosition < Text.Length)
                guiText.text += Text[currentPosition++];
            yield return new WaitForSeconds(Delay);
        }
	}
	

    void WriteText(string aText)
    {
        guiText.text = "";
        currentPosition = 0;
        Text = aText;
    }

}
